kolena/_experimental/dataset/_dataset.py

```diff
@@ -90,22 +91,18 @@ def _infer_datatype(df: pd.DataFrame) -> Union[pd.DataFrame, str]:
 
 
 def _to_serialized_dataframe(df: pd.DataFrame) -> pd.DataFrame:
-    object_columns = list(df.select_dtypes(include="object").columns)
-    result = df.select_dtypes(exclude="object")
-    result[object_columns] = df[object_columns].applymap(_serialize_dataobject)
+    result = _dataframe_object_serde(df, _serialize_dataobject)
     result[DATA_TYPE_FIELD] = _infer_datatype(df)
     result[COL_DATAPOINT] = result.to_dict("records")
-    result[COL_DATAPOINT] = result[COL_DATAPOINT].apply(lambda x: json.dumps(x, sort_keys=True))
+    result[COL_DATAPOINT] = result[COL_DATAPOINT].apply(lambda x: json.dumps(x))
```

Comment:
Does this have user-facing impact, or is object key order still lost at the persistence layer?